crates/ruff/src/args.rs

```diff
@@ -594,6 +597,38 @@ fn resolve_bool_arg(yes: bool, no: bool) -> Option<bool> {
     }
 }
 
+fn resolve_output_format_with_possibly_deprecated_conflicting_arg(
+    output_format: Option<SerializationFormat>,
+    show_sources: Option<bool>,
+) -> Option<SerializationFormat> {
+    Some(match (output_format, show_sources) {
+        (Some(o), None) => o,
+        (Some(o), Some(true)) => {
+            warn_user!("The `--show-source` argument is deprecated and has been ignored in favor of `output-format={}`.", o);
```

Comment:
Nit: consider inlining the `o` into the format string? Assuming the macro supports it.